### PR TITLE
HW-6.2

### PR DESCRIPTION
**Pull request by serg-ksv**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
...
+                    && color.equals(rectangle.color);
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 109;
+        result = 53 * result + width.hashCode();
+        result = 53 * result + length.hashCode();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Rectangle.java]: what happens, if one of the fields == null ?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
...
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        if (object.getClass().equals(Rectangle.class)) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
        if (object.getClass().equals(getClass())) {
```

Comment (serg-ksv) [reply src/main/java/core/basesyntax/Rectangle.java]: done

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (serg-ksv) [reply src/main/java/core/basesyntax/Rectangle.java]: NullPointerException

Comment (serg-ksv) [reply src/main/java/core/basesyntax/Rectangle.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**